crux_core/src/typegen.rs

```diff
@@ -69,6 +69,9 @@ use std::{
     path::{Path, PathBuf},
 };
 
+// Expose from `serde_reflection` for `register_type_with_samples()`
+pub use serde_reflection::Samples;
```

Comment:
Users have to manage the lifetime of a `Samples` object, so we need to re-export it.

Reply:
Do we need to expose `Samples` at all? My (possibly wrong) understanding is it is essentially an internal state of the tracer that may be useful in future type tracing – could we not hold it in our internal state, the same way we hold `Tracer`? The `State` type should probably have its variants named after the stages of processing, rather than what it holds (e.g. `Tracing`, `Generating`), to be a bit clearer.

Reply:
Good question. I agree it would be nice to hide `Samples`. I only exposed it to satisfy the borrow checker, because it wanted the `Samples` struct to have the same lifetime as the sample data.